staging/src/k8s.io/kubectl/pkg/drain/drain.go

```diff
@@ -101,35 +103,35 @@ type waitForDeleteParams struct {
 
 // CheckEvictionSupport uses Discovery API to find out if the server support
 // eviction subresource If support, it will return its groupVersion; Otherwise,
-// it will return an empty string
-func CheckEvictionSupport(clientset kubernetes.Interface) (string, error) {
+// it will return an empty GroupVersion
+func CheckEvictionSupport(clientset kubernetes.Interface) (schema.GroupVersion, error) {
 	discoveryClient := clientset.Discovery()
 	groupList, err := discoveryClient.ServerGroups()
 	if err != nil {
-		return "", err
+		return schema.GroupVersion{}, err
 	}
 	foundPolicyGroup := false
-	var policyGroupVersion string
 	for _, group := range groupList.Groups {
 		if group.Name == "policy" {
 			foundPolicyGroup = true
-			policyGroupVersion = group.PreferredVersion.GroupVersion
 			break
 		}
 	}
 	if !foundPolicyGroup {
-		return "", nil
+		return schema.GroupVersion{}, nil
 	}
+
+	// version info available in subresources since v1.8.0 in https://github.com/kubernetes/kubernetes/pull/49971
 	resourceList, err := discoveryClient.ServerResourcesForGroupVersion("v1")
 	if err != nil {
-		return "", err
+		return schema.GroupVersion{}, err
 	}
 	for _, resource := range resourceList.APIResources {
-		if resource.Name == EvictionSubresource && resource.Kind == EvictionKind {
-			return policyGroupVersion, nil
```

Comment:
This was incorrect before... previous versions of kubectl checked the preferred `policy` API group version and tried to set the APIVersion of the sent Eviction resource to that.

That means that when PDB graduated to policy/v1 in 1.21, kubectl started trying to set `APIVersion: "policy/v1"` in the Eviction object sent on line 168.

Fortunately (?), the typed client stomps any ApiVersion value when serializing and forced the value to be `policy/v1beta1` (fortunate, because a 1.21 API server would have choked on a "policy/v1" Eviction object)